assets/js/components/dashboard-sharing/ModuleRecoveryAlert.js

```diff
@@ -0,0 +1,229 @@
+/**
+ * ModuleRecoveryAlert component.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { Fragment, useCallback, useEffect, useState } from '@wordpress/element';
+import { sprintf, __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
+import Button from '../Button';
+import Checkbox from '../Checkbox';
+import BannerNotification from '../notifications/BannerNotification';
+import ProgressBar from '../ProgressBar';
+import Spinner from '../Spinner';
+
+const { useDispatch, useSelect } = Data;
+
+export default function ModuleRecoveryAlert() {
+	const [ checkboxes, setCheckboxes ] = useState( null );
+	const [ recoveringModules, setRecoveringModules ] = useState( false );
+
+	const recoverableModules = useSelect( ( select ) =>
+		select( CORE_MODULES ).getRecoverableModules()
+	);
+
+	const userAccessibleModules = useSelect( ( select ) => {
+		const modules = select( CORE_MODULES ).getRecoverableModules();
+
+		if ( modules === undefined ) {
+			return undefined;
+		}
+
+		const accessibleModules = Object.keys( modules ).map( ( slug ) => ( {
+			slug,
+			hasModuleAccess: select( CORE_MODULES ).hasModuleAccess( slug ),
+		} ) );
+
+		if (
+			accessibleModules.some(
+				( { hasModuleAccess } ) => hasModuleAccess === undefined
+			)
+		) {
+			return undefined;
+		}
+
+		return accessibleModules
+			.filter( ( { hasModuleAccess } ) => hasModuleAccess )
+			.map( ( { slug } ) => slug );
+	} );
+
+	const { recoverModule } = useDispatch( CORE_MODULES );
+
+	const updateCheckboxes = useCallback(
+		( slug ) =>
+			setCheckboxes( ( currentCheckboxes ) => ( {
+				...currentCheckboxes,
+				[ slug ]: ! currentCheckboxes[ slug ],
+			} ) ),
+		[]
+	);
+
+	const recoverModules = useCallback( () => {
```

Comment:
```suggestion
	const handleRecoverModules = useCallback( () => {
```
Instead of naming it `recoverModules`, `handleRecoverModules` would be a better name since it is an event handler. WDYT?